### DISCUSSION TITLE
c45459f

### DISCUSSION THREAD

**Discussion by Terracotta-6** [artisan-roaster-scope/artisan#1592] [General]
![30df75af658f8cf484e9de29b8579fc](https://github.com/artisan-roaster-scope/artisan/assets/149906183/06e2dc15-ab99-4a2a-ab38-79b2d0f2f846)

**Comment by MAKOMO** [comment]
Is this with the Windows build? Could you try with the next build again?

**Reply by Terracotta-6** [reply]
Works fine on 6d745f9. thank you ！

**Reply by MAKOMO** [reply]
Thanks for reporting. We had to downgrade numpy to 2.x again as pyinstaller (Windows/Linux only) seems to bundle 3.x correctly. We will look into this later and update to 3.x once resolved.

**Comment by MAKOMO** [comment]
Could you please retry the current build under CI construction. I had to downgrade again also the scipy package which depends on numpy. We have to wait for those package upgrades until the issue on the numpy side is resolved which might take awhile. A pity as it works well on macOS and seems to have some performance advantages. Thanks!

**Reply by Terracotta-6** [reply]
3c4e2e0 Running well, waiting to upload updates to 15203a5

**Reply by Terracotta-6** [reply]
Just roasted two boilers of coffee with the 15203a5 version and it's running great!

**Reply by MAKOMO** [reply]
Excellent! Thanks for testing that quickly. Your tests are very valuable!

**Comment by Terracotta-6** [comment]


**Reply by Terracotta-6** [reply]
Test event playback by time is faulty

**Reply by Terracotta-6** [reply]
[artisan-settings.aset.txt](https://github.com/user-attachments/files/16266040/artisan-settings.aset.txt)
[Kaleido M1-97_24-07-17_2049.alog.txt](https://github.com/user-attachments/files/16266048/Kaleido.M1-97_24-07-17_2049.alog.txt)

**Reply by MAKOMO** [reply]
What is exactly wrong here? I see that the two air events are replayed correctly, right? I cannot spot more events to be replayed. Looks all correct to me.

**Reply by Terracotta-6** [reply]
The start time of the three events cannot be started at 00:00, and can only be started after 00:00

**Reply by Terracotta-6** [reply]
The event playback of the latest version is changed. The event that starts at 00:00 cannot be played. The event must start at 00:01 to trigger the wheel cast.  When the new version is used, some events are detected that are not triggered. To trigger events after 00:00,   Alarms or other methods are required

**Reply by MAKOMO** [reply]
Very good catch! The general rule is that events are only replayed after CHARGE is set.

The condition of event replay around 00:00 has been relaxed in the latest build such that events scheduled for 00:00 in the background are replayed even if time is already passed 00:00. However, they might be replayed a little delayed at 00:01 due to switch of the feature just ON on CHARGE. If autoCHARGE is used, the CHARGE event is set in retrospect even a bit later, thus those event might only be replayed if roastime is 00:02 or 00:03, depending on your sampling rate. There is no good way around this, but better to have those events replayed a little delayed than not at all. THANKS FOR REPORTING!

**Comment by Terracotta-6** [comment]
After clicking CHARGE, it will not be displayed, but after clicking DROP, it will be displayed again. Can it be displayed during baking?This area is the Show/Hide background curve switch.A bright text display shows the background, and a darker text hides the background.
![01bd214d0f784153b52bf794b70d18e](https://github.com/user-attachments/assets/0bc75985-c607-4aca-85e3-86ac82fd1952)
![77d49012e53a747d18eeb2914bff796](https://github.com/user-attachments/assets/a490da6f-9121-43a5-b93a-3d0ed76a4646)

**Reply by MAKOMO** [reply]
<img width="659" alt="Screenshot 2024-07-21 at 01 22 44" src="https://github.com/user-attachments/assets/eafbc81c-2bb3-4c8e-9523-36bd72899b23">

**Reply by Terracotta-6** [reply]
Thank you for your answer！

**Comment by Terracotta-6** [comment]


**Reply by Terracotta-6** [reply]
Strange finding As long as SV is not equal to the number 180, the other values can be set normally.

**Reply by roasterdave** [reply]
Duplicate of Issue #1631. Your finding is correct.

**Reply by Terracotta-6** [reply]
It has been  fix , thank you!

**Comment by Terracotta-6** [comment]
![0d1c927ca638469d92542f75327d215](https://github.com/user-attachments/assets/ceae9c65-fc26-4676-a31d-59ec07a75ef8)

**Reply by roasterdave** [reply]
Thanks for finding and reporting!  Fixed in the latest build.

**Reply by Terracotta-6** [reply]
It has been fix , thank you!

**Comment by Terracotta-6** [comment]
The command ON modbus cannot transmit commands，The previous version didn't have this phenomenon, and I can't tell what the problem is. OFF The modbus command is normal.
![5f7e22290f702d32b8eb7a77d6d44d5](https://github.com/user-attachments/assets/54321499-b522-46fb-a259-6aafa52ebe88)

**Reply by MAKOMO** [reply]
That PID ON command is operational only after ON. Can you verify this?

This was already like this as early as v2.0 of Artisan (haven't checked older versions). I don't see specific reason for this restriction and thus will relax it for v3.

**Reply by Terracotta-6** [reply]
My use method is wrong, thank you for your advice.
